src/test/java/com/medved/test/TestGroupSignalChat.java

```diff
@@ -128,4 +128,64 @@ public void testGroupOfThree() throws LegacyMessageException, DuplicateMessageEx
 
 	}
 
+	@Test
+	public void testAddPersonToGroup() throws NoSessionException, InvalidMessageException, LegacyMessageException, DuplicateMessageException {
+
+		// Create Alise group session
+		InMemorySenderKeyStore aliceStore = new InMemorySenderKeyStore();
+		GroupSessionBuilder aliceSessionBuilder = new GroupSessionBuilder(aliceStore);
+		GroupCipher aliceGroupCipher = new GroupCipher(aliceStore, GROUP_SENDER_ALICE);
+		SenderKeyDistributionMessage sentAliceDistributionMessage = aliceSessionBuilder.create(GROUP_SENDER_ALICE);
+
+		// this message should be transferred via network for group initiation!
+		byte[] initGroupMessage = sentAliceDistributionMessage.serialize();
+		// also GROUP_SENDER should be transferred as well
+
+		//encript message
+		byte[] ciphertextFromAlice = aliceGroupCipher.encrypt(Alice_first_message.getBytes());
+
+		// bob receives INIT message and initiates the group session:
+		SenderKeyDistributionMessage receivedAliceDistributionMessage = new SenderKeyDistributionMessage(initGroupMessage);
+		InMemorySenderKeyStore bobStore = new InMemorySenderKeyStore();
+		GroupSessionBuilder bobSessionBuilder = new GroupSessionBuilder(bobStore);
+		GroupCipher bobGroupCipher = new GroupCipher(bobStore, GROUP_SENDER_ALICE);
+		bobSessionBuilder.process(GROUP_SENDER_ALICE, receivedAliceDistributionMessage);
+
+		// Bob decrypt and read Message from Alice
+		byte[] plaintextFromAlice = bobGroupCipher.decrypt(ciphertextFromAlice);
+		assertTrue(new String(plaintextFromAlice).equals(Alice_first_message));
+
+		//and here comes third participant - Jack
+		// initiate session for Jack
+		InMemorySenderKeyStore jackStore = new InMemorySenderKeyStore();
+		GroupSessionBuilder jackSessionBuilder = new GroupSessionBuilder(jackStore);
+		GroupCipher jackGroupCipher = new GroupCipher(jackStore, GROUP_SENDER_ALICE);
+
+		//we need to create new initial message for a group
+		SenderKeyDistributionMessage sentJackDistributionMessage = jackSessionBuilder.create(GROUP_SENDER_ALICE);
```

Comment:
I think when Jack start to send _HIS_ group initiation message he should use GROUP_SENDED_JACK in creation.
Because as i understand when Jack do like this: 
jackSessionBuilder.create(GROUP_SENDER_ALICE)

Than he will talk from Alice.